File: strip_comments.js
const fs = require('fs');
const path = require('path');

function stripCommentsJsTs(content) {
  // 1. Remove JSX comments of the form {/* ... */}
  let result = content.replace(/\{\s*\/\*[\s\S]*?\*\/\s*\}/g, '');
  
  // 2. Character-by-character state machine to remove JSDoc, multi-line, and single-line comments
  let output = [];
  let i = 0;
  const n = result.length;
  let state = "normal";
  let inString = null;
  let escape = false;
  
  while (i < n) {
    const char = result[i];
    
    if (state === "normal") {
      if (i + 1 < n && result.slice(i, i + 2) === "//") {
        state = "single_line_comment";
        i += 2;
        continue;
      } else if (i + 1 < n && result.slice(i, i + 2) === "/*") {
        state = "multi_line_comment";
        i += 2;
        continue;
      } else if (char === '"' || char === "'" || char === "`") {
        state = "string";
        inString = char;
        escape = false;
        output.push(char);
      } else {
        output.push(char);
      }
    } else if (state === "string") {
      output.push(char);
      if (escape) {
        escape = false;
      } else if (char === "\\") {
        escape = true;
      } else if (char === inString) {
        state = "normal";
        inString = null;
      }
    } else if (state === "single_line_comment") {
      if (char === "\n") {
        state = "normal";
        output.push("\n");
      }
    } else if (state === "multi_line_comment") {
      if (i + 1 < n && result.slice(i, i + 2) === "*/") {
        state = "normal";
        i += 2;
        continue;
      }
    }
    i++;
  }
  
  let finalResult = output.join('');
  
  // Remove empty curly braces {} that might be left in JSX after stripping comments
  finalResult = finalResult.replace(/\{\s*\}/g, '');
  
  return finalResult;
}

function stripCommentsCss(content) {
  let output = [];
  let i = 0;
  const n = content.length;
  let state = "normal";
  let inString = null;
  let escape = false;
  
  while (i < n) {
    const char = content[i];
    
    if (state === "normal") {
      if (i + 1 < n && content.slice(i, i + 2) === "/*") {
        state = "multi_line_comment";
        i += 2;
        continue;
      } else if (char === '"' || char === "'") {
        state = "string";
        inString = char;
        escape = false;
        output.push(char);
      } else {
        output.push(char);
      }
    } else if (state === "string") {
      output.push(char);
      if (escape) {
        escape = false;
      } else if (char === "\\") {
        escape = true;
      } else if (char === inString) {
        state = "normal";
        inString = null;
      }
    } else if (state === "multi_line_comment") {
      if (i + 1 < n && content.slice(i, i + 2) === "*/") {
        state = "normal";
        i += 2;
        continue;
      }
    }
    i++;
  }
  
  return output.join('');
}

function processFile(filepath) {
  console.log(`Processing: ${filepath}`);
  const content = fs.readFileSync(filepath, 'utf8');
  const ext = path.extname(filepath);
  
  let newContent;
  if (['.ts', '.tsx', '.js', '.jsx'].includes(ext)) {
    newContent = stripCommentsJsTs(content);
  } else if (ext === '.css') {
    newContent = stripCommentsCss(content);
  } else {
    return;
  }
  
  fs.writeFileSync(filepath, newContent, 'utf8');
  console.log(`Finished: ${filepath}`);
}

function walkAndStrip(dir) {
  if (!fs.existsSync(dir)) return;
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      walkAndStrip(fullPath);
    } else {
      const ext = path.extname(fullPath);
      if (['.ts', '.tsx', '.js', '.jsx', '.css'].includes(ext)) {
        processFile(fullPath);
      }
    }
  }
}

// Strip comments in app/ and lib/
console.log("Starting comment stripping...");
walkAndStrip('app');
walkAndStrip('lib');
console.log("All comments stripped successfully!");

File: strip_comments.py
import os
import re

def strip_comments_js_ts(content):
    output = []
    i = 0
    n = len(content)
    state = "normal"
    in_string = None
    escape = False
    
    while i < n:
        char = content[i]
        
        if state == "normal":
            if i + 1 < n and content[i:i+2] == "//":
                state = "single_line_comment"
                i += 2
                continue
            elif i + 1 < n and content[i:i+2] == "/*":
                state = "multi_line_comment"
                i += 2
                continue
            elif char in ('"', "'", "`"):
                state = "string"
                in_string = char
                escape = False
                output.append(char)
            else:
                output.append(char)
        
        elif state == "string":
            output.append(char)
            if escape:
                escape = False
            elif char == "\\":
                escape = True
            elif char == in_string:
                state = "normal"
                in_string = None
                
        elif state == "single_line_comment":
            if char == "\n":
                state = "normal"
                output.append("\n")
                
        elif state == "multi_line_comment":
            if i + 1 < n and content[i:i+2] == "*/":
                state = "normal"
                i += 2
                continue
                
        i += 1
        
    result = "".join(output)
    
    # Post-process to remove empty JSX braces left over from {/* comments */}
    # Matches { } with any whitespace inside
    result = re.sub(r'\{\s*\}', '', result)
    
    return result

def strip_comments_css(content):
    output = []
    i = 0
    n = len(content)
    state = "normal"
    in_string = None
    escape = False
    
    while i < n:
        char = content[i]
        
        if state == "normal":
            if i + 1 < n and content[i:i+2] == "/*":
                state = "multi_line_comment"
                i += 2
                continue
            elif char in ('"', "'"):
                state = "string"
                in_string = char
                escape = False
                output.append(char)
            else:
                output.append(char)
                
        elif state == "string":
            output.append(char)
            if escape:
                escape = False
            elif char == "\\":
                escape = True
            elif char == in_string:
                state = "normal"
                in_string = None
                
        elif state == "multi_line_comment":
            if i + 1 < n and content[i:i+2] == "*/":
                state = "normal"
                i += 2
                continue
                
        i += 1
        
    return "".join(output)

def process_file(filepath):
    print(f"Start processing: {filepath} ...")
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
        
    ext = os.path.splitext(filepath)[1]
    if ext in ('.ts', '.tsx', '.js', '.jsx'):
        new_content = strip_comments_js_ts(content)
    elif ext == '.css':
        new_content = strip_comments_css(content)
    else:
        return
        
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(new_content)
    print(f"Finished processing: {filepath}")

def walk_and_strip():
    target_dirs = ['app', 'lib']
    for target in target_dirs:
        for root, dirs, files in os.walk(target):
            for file in files:
                ext = os.path.splitext(file)[1]
                if ext in ('.ts', '.tsx', '.js', '.jsx', '.css'):
                    filepath = os.path.join(root, file)
                    process_file(filepath)

if __name__ == "__main__":
    walk_and_strip()
    print("Comment stripping complete!")
